src/OrchardCore/OrchardCore/Shell/ShellHost.cs

```diff
@@ -521,6 +521,18 @@ public void Dispose()
             {
                 shell.Dispose();
             }
+
+            GC.SuppressFinalize(this);
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            foreach (var shell in ListShellContexts())
+            {
+                await shell.DisposeAsync();
+            }
+
+            GC.SuppressFinalize(this);
```

Comment:
Is there a finalizer in this class?

Reply:
No but just followed the analyzer suggestion CA1816:

> To prevent derived types introducing finalizers from having to reimplement IDisposable and to call it, unsealed types without finalizers should still call GC.SuppressFinalize.

But I can remove it if you prefer and maybe make ShellHost a sealed class.

Reply:
Okay I removed the `GC.SuppressFinalize()` and made `ShellHost` a sealed class.